pkg/apis/project/basic.go

```diff
@@ -125,10 +126,18 @@ func (h Handler) CollectionDelete(req CollectionDeleteRequest) error {
 	// Validate whether the subject has permission to delete the projects.
 	sj := session.MustGetSubject(req.Context)
 	if !sj.IsAdmin() {
-		for i := range ids {
-			if !sj.Enforce(string(ids[i]), http.MethodDelete, "projects", string(ids[i]), req.Context.FullPath()) {
-				return errorx.NewHttpError(http.StatusForbidden, "")
-			}
+		cnt, err := h.modelClient.SubjectRoleRelationships().Query().
+			Where(
+				subjectrolerelationship.ProjectIDIn(ids...),
+				subjectrolerelationship.RoleID(types.ProjectRoleOwner)).
```

Comment:
What's the problem of previous implementation? Is it possible that we have custom role that can also delete project in the future?

Reply:
if so, we should prevent that, and make sure only the `project/owner` can delete the project.